lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -942,7 +942,7 @@ def add_header(file_accessor, build_file, public_headers, private_headers, nativ
                     'Project'
                   end
 
-            if target.build_as_framework? && any_header_mapping_dirs? && acl != 'Project'
+            if target.build_as_framework? && header_mappings_dir(file_accessor) != nil && acl != 'Project'
```

Comment:
this is the fix. Do not add a copy headers phase unless there is a header mappings dir specified for the header.

This means that if a subspecs has a header mappings dir but the root spec doesn't we would get build errors with the generated umbrella header not pointing to the correct paths.